dll/win32/kernel32/winnls/string/lang.c

```diff
@@ -2008,6 +2008,37 @@ INT WINAPI LCMapStringEx(LPCWSTR name, DWORD flags, LPCWSTR src, INT srclen, LPW
         }
     }
 
+#ifdef __REACTOS__
+    if (flags & LCMAP_KATAKANA)
+    {
+        INT convlen = dst_ptr - dst;
+        for (dst_ptr = dst; convlen; --convlen, ++dst_ptr)
+        {
+            WCHAR wch = *dst_ptr;
+            if (0x3041 <= wch && wch <= 0x3093) /* U+3041 ... U+3093: Hiragana */
+                *dst_ptr = wch + 0x60; /* Hiragana to Katanaka */
+            else if (wch == 0x3095) /* U+3095: HIRAGANA LETTER SMALL KA */
+                *dst_ptr = 0x30F5; /* U+30F5: KATAKANA LETTER SMALL KA */
+            else if (wch == 0x309D) /* U+309D: Hiragana Iteration Mark */
+                *dst_ptr = 0x30FD; /* U+30FD: Katakana Iteration Mark */
+        }
+    }
+    else if (flags & LCMAP_HIRAGANA)
+    {
+        INT convlen = dst_ptr - dst;
+        for (dst_ptr = dst; convlen; --convlen, ++dst_ptr)
+        {
+            WCHAR wch = *dst_ptr;
+            if (0x30A1 <= wch && wch <= 0x30F3) /* U+30A1 ... U+30F3: Katakana */
+                *dst_ptr = wch - 0x60; /* Katanaka to Hiragana */
+            else if (wch == 0x30F5) /* U+30F5: KATAKANA LETTER SMALL KA */
+                *dst_ptr = 0x3095; /* U+3095: HIRAGANA LETTER SMALL KA */
+            else if (wch == 0x30FD) /* U+30FD: Katakana Iteration Mark */
+                *dst_ptr = 0x309D; /* U+309D: Hiragana Iteration Mark */
```

Comment:
It seems that for those two characters there is also a subtraction by 0x60 being done.
Couldn't this be grouped as follows? :
```c
else if ((wch == 0x30F5) || wch == 0x30FD)
    *dst_ptr = wch - 0x60;
```
same question also for the other case above, where instead we instead have an addition of 0x60.

Reply:
Fixed.